src/main/java/mobile/android/ProductsPage.java

```diff
@@ -0,0 +1,111 @@
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mobile.common.CartPageBase;
+import mobile.common.ProductsPageBase;
+import mobile.common.SortProductsPageBase;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+@DeviceType(pageType = Type.ANDROID_PHONE, parentClass = ProductsPageBase.class)
+public class ProductsPage extends ProductsPageBase implements IMobileUtils {
+
+    @FindBy(xpath = "//android.widget.TextView[@content-desc='test-Item title']")
+    protected ExtendedWebElement productTitles;
+
+    @FindBy(xpath = "//android.widget.TextView[@text='%s']"
+        + "/following-sibling::android.widget.TextView[@content-desc='test-Price']")
+    protected ExtendedWebElement productPrice;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-Cart']")
+    private ExtendedWebElement cartButton;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-Modal Selector Button']")
+    private ExtendedWebElement sortButton;
+
+    @FindBy(xpath = "//android.widget.TextView["
+        + "@content-desc='test-Item title' and @text='%s']"
+        + "/following-sibling::android.view.ViewGroup[contains(@content-desc, 'test-ADD TO CART')]")
+    private ExtendedWebElement addToCartButton;
+
+    @FindBy(xpath = "//android.view.ViewGroup[@content-desc='test-Cart']"
+        + "/android.view.ViewGroup/android.widget.TextView")
+    private ExtendedWebElement cartCount;
+
+    @FindBy(xpath = "//android.widget.TextView[contains(@text, 'Terms of Service')]")
+    private ExtendedWebElement termsOfService;
+
+    public ProductsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public boolean isOpen() {
+        return sortButton.isElementPresent();
+    }
+
+    @Override
+    public void addProductToCartByName(String name) {
+        swipe(addToCartButton.format(name), Direction.UP);
+        addToCartButton.format(name).click();
+    }
+
+    @Override
+    public boolean isProductPresent(String name) {
+        return swipe(addToCartButton.format(name), Direction.UP);
+    }
+
+    @Override
+    public int getCartCount() {
+        String strCount = cartCount.getText();
+        return Integer.parseInt(strCount);
+    }
+
+    @Override
+    public CartPageBase clickCartButton() {
+        cartButton.click();
+        return initPage(CartPageBase.class);
+    }
+
+    @Override
+    public SortProductsPageBase clickSortButton() {
+        sortButton.click();
+        return initPage(SortProductsPageBase.class);
+    }
+
+    @Override
+    public Map<String, Double> getProductTitlePrices() {
+        Map<String, Double> titlePricesMap = new LinkedHashMap<>();
+        boolean found = false;
+        while (!found) {
+            found = swipe(termsOfService, 1);
+            List<String> productTitlesDisplayed = productTitles.formatToList().stream()
```

Comment:
How does productTitles hold multiple objects to make it into a list?

Reply:
In this method under the products screen, each display of the screen shows between two to four product titles depending on where the swipe lands

Reply:
@jinocencio-solvd Then can we use List<ExtendedWebElement> for productTitles ?

Reply:
Yes I can declare `productTitles` as `List<ExtendedWebElement> ` in the fields instead of perforing the operation `productTitles.formatToList()`.  This way would be more clear.

Reply:
I understand it now. Thank you very much.